### PR TITLE
Stricter idempotent import executor api

### PR DESCRIPTION
**Pull request by xokker**

### CODE REVIEW

**Comment by seehamrun** [comment]
This PR has a lot of moving pieces. Can you break up the change so that it contains the main changes across the IdempotentExcutor only and then send a new PR for the changes across the Importers?

**Comment by xokker** [comment]
@seehamrun makes perfect sense, thank you! I reverted the non-essential changes and will re-apply them as a new PR when this one is merged.

File: portability-types-common/src/main/java/org/datatransferproject/types/common/models/photos/PhotoModel.java
Comment (jzacsh) [inline portability-types-common/src/main/java/org/datatransferproject/types/common/models/photos/PhotoModel.java:29]: (shouldn't block this PR) is it worth filing a bug to go update all the models to implement this?

File: portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/idempotentexecutor/ItemImportResult.java
Comment (jzacsh) [inline portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/idempotentexecutor/ItemImportResult.java:9]:
```suggestion
   * Internal cache of successfully imported data, or Null in case of failure.
```

**Review by jzacsh — DISMISSED** [review]

**Comment by jzacsh** [comment]
I left some comments, but please definitely merge to unblock yourself, and feel free to fix/address these things later.

File: portability-types-common/src/main/java/org/datatransferproject/types/common/models/photos/PhotoModel.java
Comment (wmorland) [reply portability-types-common/src/main/java/org/datatransferproject/types/common/models/photos/PhotoModel.java:29]: Good suggestion. Filed #1094

**Review by wmorland — APPROVED** [review]
Approve new version with suggested changes

**Review by wmorland — APPROVED** [review]
Fixed CLA issue from co-author

**Pull request merged**
